### PR TITLE
Auth: Add fixed repeat migration w. checks for subset of permissions

### PR DESCRIPTION
**Pull request by eleijonmarck**
Co-authored-by: Karl Persson <kalle.persson@grafana.com>

**What is this feature?**
We have missing permissions for `alert_rules` which leads to missing folder permissions in the frontend, as managed permission service won't correctly translate between RBAC permissions and "user friendly" permissions due to missing alerting RBAC permissions.

This PR fixes a version of the `AddManagedFolderAlertActionsRepeatMigration` migration to fix this issue. It ensures that alerting permissions that have not been added will be added correctly.

**Which issue(s) does this PR fix?**:
https://github.com/grafana/grafana-enterprise/issues/4084

support ticket where issue originated from - https://github.com/grafana/support-escalations/issues/4277

**Special notes for your reviewer**:

Ran on a replicated issue and worked.
Have tested on a copy where we delete an explicit row of `alert_rules:read` permission missing.

image

### CODE REVIEW

**Comment by eleijonmarck** [comment]
@kalleep before reviewing, i will test this on a replicated issue first.

File: pkg/services/sqlstore/migrations/accesscontrol/dashboard_permissions.go [hunk 1]
```diff
@@ -408,6 +408,15 @@ func AddManagedFolderAlertActionsRepeatMigration(mg *migrator.Migrator) {
 	mg.AddMigration(managedFolderAlertActionsRepeatMigratorID, &managedFolderAlertActionsRepeatMigrator{})
 }
 
+const managedFolderAlertActionsRepeatMigratorFixedID = "managed folder permissions alert actions repeated fixed migration"
+
+/*
+AddManagedFolderAlertActionsRepeatFixedMigration is a fixed version of AddManagedFolderAlertActionsRepeatMigration.
+*/
+func AddManagedFolderAlertActionsRepeatFixedMigration(mg *migrator.Migrator) {
```
Comment (IevaVasiljeva) [inline pkg/services/sqlstore/migrations/accesscontrol/dashboard_permissions.go:416]: I think this function should be called from somewhere here: https://github.com/grafana/grafana/blob/main/pkg/services/sqlstore/migrations/migrations.go#L106

**Review by IevaVasiljeva — APPROVED** [review]
Thanks for spotting this and fixing it! I've added a comment inline, but apart from it the changes look good.

I've thought about this for a while now, and can't see any problems with this fixed version. So fingers crossed we've got it right this time! 🤞

File: pkg/services/sqlstore/migrations/accesscontrol/dashboard_permissions.go (hunk 1 above)
Comment (eleijonmarck) [reply pkg/services/sqlstore/migrations/accesscontrol/dashboard_permissions.go:416]: sorry had that checked in previously, missed to check in after some changes

Comment (eleijonmarck) [reply pkg/services/sqlstore/migrations/accesscontrol/dashboard_permissions.go:416]: https://github.com/grafana/grafana/pull/58054/files#diff-0fd985df1b8aaadcea6def96014b31f7eee27c4ea19761733754bffcf7d06f6aR106

**Pull request merged**